app/survey_translator.py

```diff
@@ -0,0 +1,237 @@
+                    msgctxt = line.split('"')[1]
+                elif line.startswith('msgid'):
+                    value = line.split('"')[1]
+                elif line.startswith('msgstr'):
+                    object_key = self.get_object_key(value, msgctxt)
+                    if value != "":
+                        translation = line.split('"')[1]
+                        self.translation_object[object_key] = dumb_to_smart_quotes(translation)
+                    msgctxt = None
+
+    def get_object_key(self, value, msgctxt):
+        return (TemplateExtractor.trim_chars_from_object_key(value),
+                TemplateExtractor.trim_chars_from_object_key(msgctxt))
+
+    def validate_translation_object(self, translation_object):
+        """
+        Identify missing translation values within the SurveyTranslator
+        for a given template
+
+        :param translation_object:
+        :return:
+        """
+        for object_key in translation_object.keys():
+            if self.translation_object.get(object_key) is None:
+                missing_object = translation_object[object_key]
+                error_message = 'Missing translation for '
+                id = missing_object['context'][0]
+                for context in missing_object['context']:
+                    if 'id: ' in context:
+                        id = 'id:' + context.split(':')[1]
+                error_message += id + ' ~ "' + missing_object['value'] + '"'
+                self.missing_translations += [error_message]
+                print(error_message)
+
+    def translate_schema(self, data):
+        if 'legal_basis' in data and data['legal_basis'] != 'Voluntary':
+            data['legal_basis'] = self.get_translation_value(data['legal_basis'])
+
+        # Now build up translatable text from the nested dictionaries and lists
+        for section in data['sections']:
+            self.write_text_for_container(section)
+
+            for group in section['groups']:
+                self.write_text_for_container(group)
+
+                for block in group['blocks']:
+                    self.write_text_for_container(block)
+
+                    if 'questions' not in block:
+                        self.get_non_question_translatable_text(block)
+
+                    for question in block.get('questions', {}):
+                        msgctxt = 'For Question: ' + self.get_title_text_for_msgctxt(question)
+
+                        self.write_question_translatable_text(question, msgctxt)
+
+                        for answer in question.get('answers', []):
+                            self.write_text_for_container(answer, msgctxt)
+                            self.get_guidance_text(answer, msgctxt)
+                            self.get_options_text(answer, msgctxt)
+                            self.get_validation_text(answer, msgctxt)
+
+        return data
+
+    def get_title_text_for_msgctxt(self, container):
```

Comment:
Minor, but this method name implies it's getting the title text _for_ a given msgctxt. This would be clearer as `get_title_text_for_container`, or just `get_title_text`.